guides/runbooks/orca-zombie-executions.md

```diff
@@ -0,0 +1,118 @@
+You can attempt to re-hydrate the queue --- reissue messages onto the work queue based on the last stored state --- using an [admin API in Orca](https://github.com/spinnaker/orca/blob/master/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/admin/web/QueueAdminController.kt#L33), which must be called directly as it is not exposed through Gate.
+This command can take either a single execution or operate on all executions within a time range. 
+**This command will dry-run by default.**
+To actually rehydrate the queue, pass the query parameter `dryRun=false`.
+
+```bash
+$ curl -XPOST \
+  https://localhost:8083/admin/queue/hydrate?executionId=01CS076X85RX6MWBTQ0VGBF8VX?dryRun=false
+```
+
+This command is **best effort** and may not be able to rehydrate the Execution, especially if the Execution was zombied while running a non-retryable task.
+
+An example response from the endpoint:
+
+```json
+{
+  "dryRun": false,
+  "executions": {
+    "01CS076X85RX6MWBTQ0VGBF8VX": {
+      "startTime": 1538679600852,
+      "actions": [
+        {
+          "description": "Task is running and is retryable",
+          "message": {
+            "kind": "runTask",
+            "executionType": "PIPELINE",
+            "executionId": "01CS076X85RX6MWBTQ0VGBF8VX",
+            "application": "myapplication",
+            "stageId": "01CS076X8501MNAD2ZTJ4ST2TM",
+            "taskId": "1",
+            "taskType": "com.netflix.spinnaker.orca.echo.pipeline.ManualJudgmentStage$WaitForManualJudgmentTask",
+            "attributes": [],
+            "ackTimeoutMs": 600000
+          },
+          "context": {
+            "stageId": "01CS076X8501MNAD2ZTJ4ST2TM",
+            "stageType": "manualJudgment",
+            "stageStartTime": 1538682406227,
+            "taskId": "1",
+            "taskType": "waitForJudgment",
+            "taskStartTime": 1538682406242
+          }
+        },
+        {
+          "description": "Task is running but is not retryable",
+          "context": {
+            "stageId": "01CS076X85ECXHF3FRWZBTQ359",
+            "stageType": "createProperty",
+            "stageStartTime": 1538681485559,
+            "taskId": "3",
+            "taskType": "monitorProperties",
+            "taskStartTime": 1538681546116
+          }
+        }
+      ],
+      "canApply": false
+    }
+  }
+}
+```
+
+For each Execution, a final action summary is provided `canApply`. 
+If any part of an Execution cannot be re-hydrated, the entire Execution will be skipped.
+
+## Cancel the Execution
+
+If the Execution cannot be rehydrated, it will need to be canceled. You can cancel the Execution via the UI.
+
+## Known Causes
+
+Zombie Executions can occur due to a loss of the Redis instance backing the Orca queue, or in prolonged unreliable networks.
+If you're using the `RedisExecutionRepository`, it's likely that when you lose the Redis backing the queue, you're also losing all running Executions.
```

Comment:
Can "using the `RedisExecutionRepository`" be linked to anything? Do you know if we describe that anywhere? I don't find any doc for `RedisExecutionRepository`.

Reply:
We don't have any documentation for the RedisExecutionRepository. It's the default if SQL hasn't been configured.